src/api/app/controllers/webui/repositories_controller.rb

```diff
@@ -148,14 +148,16 @@ def create_dod_repository
   # POST project/create_image_repository
   def create_image_repository
     authorize @project, :update?
-    repository = @project.repositories.create!(name: 'images')
-    Architecture.available.each do |architecture|
-      repository.repository_architectures.create!(architecture: architecture)
-    end
-    @project.prepend_kiwi_config
+    repository = @project.repositories.new(name: 'images')
 
     if repository.save
+      Architecture.available.each do |architecture|
```

Comment:
Wouldn't this also raise exceptions (haven't checked the validations in RepositoryArchitectures ...). At least it would create the architectures again every time, right?

Reply:
Can you please elaborate on your response, I don't understand it ... :/

Reply:
Talked in person about it! Should be fine as a new repository has no architectures.